src/staticanalysis/bot/static_analysis_bot/coverity/coverity.py

```diff
@@ -180,7 +180,17 @@ def return_issues(self, coverity_results_path, revision):
                 logger.info('Coverity did not find any issues')
                 return []
 
-            return [CoverityIssue(issue, revision) for issue in result['issues']]
+            issues = []
+            for issue in result['issues']:
+                stateOnServer = issue['stateOnServer']
+                # According to Coverity manual:
+                # presentInReferenceSnapshot - True if the issue is present in the reference
+                # snapshot specified in the cov-run-desktop command, false if not.
+                if stateOnServer is not None and 'presentInReferenceSnapshot' in stateOnServer and stateOnServer[
```

Comment:
It would be cleaner to define a local function `_isOnServer(issue)`, such as:

```python
def _isOnServer(issue):
  stateOnServer = issue['stateOnServer']
  return your long long criteria
```

So that you can do:

```python
return [
  CoverityIssue(issue, revision)
  for issue is result['issues']
  if _isOnServer(issue)
]
```

This way if you have more test cases later on, it's easier to keep track

Reply:
This sounds very reasonable.